tests/bench/harness.py

```
@@ -149,6 +157,99 @@ def start_database(self) -> None:
+
+        while time.time() - start_time < timeout:
+            try:
+                import redis
+
+                # Test Redis connection
+                r = redis.Redis(
+                    host="localhost", port=redis_port, decode_responses=True
+                )
+                r.ping()
+                print("Redis is ready!")
+                return True
```

Ruff (0.14.1)

Consider moving this statement to an `else` block

(TRY300)